pkg/agent/clean/adunmigration/rtbs.go

```diff
@@ -390,6 +465,26 @@ func migrateGRBs(workunit *migrateUserWorkUnit, sc *config.ScaledContext, dryRun
 				}
 				return true, nil
 			})
+			if err != nil {
+				logrus.Errorf("[%v] permanent error when deleting GRB, giving up: %v", migrateGrbsOperation, err)
+				continue
+			}
+
+			// Adding the annotation here has the side effect of triggering a downstream RBAC sync for our newly created
+			// binding, which is why we perform it last
+			newAnnotations[adGUIDMigrationAnnotation] = oldGrb.UserName
+			err = wait.ExponentialBackoff(backoff, func() (finished bool, err error) {
+				createdGrb.Annotations = newAnnotations
```

Comment:
I'd recommend re-fetching the GRB here before updating like you do with the CRTB/PRTB.